packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -148,18 +163,28 @@ class AnalyzeCommand extends FlutterCommand {
 
     if (argResults['current-package']) {
       // ./lib/main.dart
-      String mainPath = 'lib/main.dart';
+      String mainPath = path.join('lib', 'main.dart');
```

Comment:
if this is a change that we need to do, then we should do it throughout the flutter tool. I doubt this is the only place that does it.